### PR TITLE
BLAST+ 2.7.0

### PR DESCRIPTION
**Pull request by peterjc**
Updates the recipe from BLAST+ 2.6.0 to 2.7.0

Also switched from HTTP to HTTPS in line with NCBI site changes.

Also switched from MD5 to SHA256 checksum.

* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [X] This PR does something else (explain below).

### CODE REVIEW

File: recipes/blast/meta.yaml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
     patches:
       - boost_106400.patch # [linux]
     fn: ncbi-blast-{{ version }}+-x64-macosx.tar.gz # [osx]
-    url: http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/{{ version }}/ncbi-blast-{{ version }}+-x64-macosx.tar.gz # [osx]
-    md5: 7dc2ebc86b3a064bae6a3e378495d414 # [osx]
+    url: https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/{{ version }}/ncbi-blast-{{ version }}+-x64-macosx.tar.gz # [osx]
+    sha256: c295aa858638a8194d6b4abbefedaa2759e67ee2932e3553ffbc4afa138b20ee # [osx]
 
 build:
   number: 2
```
Comment (nsoranzo) [inline recipes/blast/meta.yaml]: Reset the build number to 0.

Comment (peterjc) [reply recipes/blast/meta.yaml]: Thanks - I was meaning to do that but forgot once at the command line.

**Comment by peterjc** [comment]
Green light from TravisCI - if no one objects I'll merge this later today.

**Comment by nsoranzo** [comment]
Thanks @peterjc!

**Comment by peterjc** [comment]
Thanks :)

**Pull request merged**